docs/architecture/grpc-for-wcf-developers/channel-credentials.md

```diff
@@ -0,0 +1,153 @@
+
+### Adding ASP.NET Core Certificate authentication
+
+Certificate authentication is provided by the Microsoft.AspNetCore.Authentication.Certificate package ([www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Certificate)).
+
+Add the Certificate authentication service in `ConfigureServices`, and add Authentication and Authorization to the ASP.NET Core pipeline in `Configure`.
+
+```csharp
+public class Startup
+{
+    private readonly bool _isDevelopment;
+
+    public Startup(IWebHostEnvironment env)
+    {
+        _isDevelopment = env.IsDevelopment();
+    }
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddAuthentication(CertificateAuthenticationDefaults.AuthenticationScheme)
+            .AddCertificate(options =>
+            {
+                if (_isDevelopment)
+                {
+                    // DO NOT DO THIS IN PRODUCTION!
+                    options.RevocationMode = X509RevocationMode.NoCheck;
+                }
+            });
+        services.AddAuthorization();
+        services.AddGrpc();
+    }
+
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseAuthentication();
+        app.UseAuthorization();
+
+        app.UseRouting();
+
+        app.UseEndpoints(endpoints => { endpoints.MapGrpcService<GreeterService>(); });
+    }
+
+}
+```
+
+## Using a certificate with the client
+
+With the Grpc.Net.Client package, certificates are configured on an `HttpClient` instance that is provided to the `GrpcChannel` used for the connection.
+
+```csharp
+class Program
+{
+    static async Task Main(string[] args)
+    {
+        // Assume path to client .pfx file and password are passed from command line
+        var cert = new X509Certificate2(args[0], args[1]);
+
+        var handler = new HttpClientHandler();
+        handler.ClientCertificates.Add(cert);
+        var httpClient = new HttpClient(handler);
+
+        var channel = GrpcChannel.ForAddress("https://localhost:5001/", new GrpcChannelOptions
+        {
+            HttpClient = httpClient
+        });
+
+        var grpc = new Greeter.GreeterClient(channel);
+        var response = await grpc.SayHelloAsync(new HelloRequest { Name = "Bob" });
+        System.Console.WriteLine(response.Message);
+    }
+}
+```
+
+## Combining ChannelCredentials and CallCredentials
+
+You can configure your server to use both certificate and token authentication by applying the certificate changes to the Kestrel server and using the JWT bearer middleware in ASP.NET Core.
+
+To provide both ChannelCredentials and CallCredentials on the client, use the `ChannelCredentials.Create` method to apply the call credentials. Certificate authentication still needs to be applied using the `HttpClient` instance: if you pass any arguments to the `SslCredentials` constructor the internal client code will throw an exception.
```

Comment:
SslCredentials doesn't do anything in Grpc.Net.Client other than verify that you are calling a HTTPS address. It only exists to provide type compatibility with Grpc.Core.